File: 412-FizzBuzz/412-FizzBuzz.js
// Last updated: 3/6/2026, 2:09:17 PM
/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function (n) {
    let res = [];
    for (let i = n; i >= 1; i--) {
        if (i % 3 === 0 && i % 5 === 0) {
            res.unshift("FizzBuzz");
        }
        else if (i % 3 === 0) {
            res.unshift("Fizz");
        }
        else if (i % 5 === 0) {
            res.unshift("Buzz");
        }
        else {
            res.unshift(String(i));
        }
    }
    return res;
};

File: 412-FizzBuzz/412-FizzBuzz.cpp
// Last updated: 3/27/2026, 5:12:38 PM
1class Solution {
2public:
3    vector<string> fizzBuzz(int n) {
4        vector<string> answer;
5
6        for(int i=1;i<=n;i++){
7            if(i%3==0 && i%5==0){
8                answer.push_back("FizzBuzz");
9            }
10            else if(i%3==0){
11                answer.push_back("Fizz");
12            }
13            else if(i%5==0){
14                answer.push_back("Buzz");
15            }
16            else{
17                answer.push_back(to_string(i));
18            }
19        }
20        return answer;
21    }
22};